main_test.go

```diff
@@ -1,11 +1,29 @@
 package main
 
-import "testing"
+import (
+	"testing"
+	"time"
+)
 
 func Test_main(t *testing.T) {
+	go func() {
+		time.Sleep(2 * time.Second)
+		httpServer.Close()
+		httpsServer.Close()
+	}()
```

Comment:
we should probably look into testing/synctest in go 1.24 and try to enable it 

https://go.dev/blog/synctest

We can do so by setting an environment variable GOEXPERIMENT=synctest

In the test:
```
synctest.Run(func(){

})
```

This will do so the test dont need to sleep and maybe not need the close on http servers